README.md

```diff
@@ -16,80 +16,166 @@ These steps are detailed in the [Google Cloud documentation](https://cloud.googl
+```
 
-The styling will be broken, but you should be able to log in to the Wagtail admin interface at http://127.0.0.1:8000/admin/ with the user details you have just created.
+Configure CORS headers on the static bucket so fonts can be loaded from the
+site's domain, and make the contents of both buckets readable by default:
 
-## Create a bucket for static content
+```
+gsutil cors set cors.json gs://my-wagtail-site-static
+gsutil defacl set public-read gs://my-wagtail-site-static
+gsutil defacl set public-read gs://my-wagtail-site-media
+```
+
+## Configure local settings
+
+Create a file called `wagae/settings/local.py`, and add your site-specific
+settings:
 
 ```
-gsutil mb gs://<GCS BUCKET NAME>
-gsutil defacl set public-read gs://<GCS BUCKET NAME>
-gsutil cors set cors.json gs://<GCS BUCKET NAME>
+DB_CONNECTION_NAME = '[CONNECTION_NAME]'
+DB_PASSWORD = '[DB_PASSWORD]'
+GS_BUCKET_PREFIX = '[BUCKET_PREFIX]'
+GS_PROJECT_ID = '[GCP_PROJECT_ID]'
+SECRET_KEY = '[SECRET_KEY]'
 ```
 
-Update `STATIC_URL` in `wagae/settings/base.py`:
+`[GCP_PROJECT_ID]` is the name of your GCP project.  `[BUCKET_PREFIX]` is the
+common prefix of the two GCS buckets you created, e.g. `my-wagtail-site`.
+
+`[SECRET_KEY]` should be a long string of random bytes, e.g. generated by
+`pwgen -s 64 1`.
+
+## Install the database tables
 
-```python
-STATIC_URL = 'https://storage.googleapis.com/<GCS BUCKET NAME>/static/'
+1. Run Cloud SQL Proxy in a separate terminal:
+
+`./cloud_sql_proxy -instances="[CONNECTION_NAME]=tcp:5432"`
+
+2. Create Wagtail's database tables and an initial user:
+
+```
+./manage.py migrate
+./manage.py createsuperuser
 ```
 
-and copy the files across:
+## Upload static files
+
+Ideally, we would let `django-storages` upload static files for us, but
+currently it doesn't seem to handle directory paths properly, which breaks asset
+loading.  Instead, collect the static files locally and upload them by hand:
 
 ```
-gsutil rsync -R static/ gs://<GCS BUCKET NAME>/static
+./manage.py collectstatic --noinput
+gsutil -m rsync -R static/ gs://my-wagtailsite-static/
 ```
 
-## Configure the app for App Engine
+## Check everything works locally
+
+`./manage.py runserver`
+
+You should be able to log in to the Wagtail admin interface at
+http://127.0.0.1:8000/admin/ with the user details you have just created.
 
-Set `<INSTANCE CONNECTION NAME>` in `app.yaml`.
+## Deploy to App Engine
 
-In `settings/base.py`, update `<GCS BUCKET NAME>` and `<GCP PROJECT ID>`, 
-and set `<YOUR SECRET KEY>`, e.g. using
+Edit `app.yaml` and set `[CONNECTION_NAME]` to the Cloud SQL database connection
+name.
 
-```python
-import random
-''.join([random.SystemRandom().choice(
-    'abcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*(-_=+)'
-    ) for i in range(50)])
+Then deploy the application:
+
+```
+gcloud app deploy
 ```
 
-Deploy your site with `gcloud app deploy`.
+## Set storage permissions
+
+Now that App Engine has created the service account for the application, set the
```

Comment:
Why didn't I need to do this?

Reply:
I don't know.  Did you not get a permission denied error when you tried to upload images or documents?

Reply:
No